core/controller/cluster/controller_operation.go

```diff
@@ -702,3 +727,60 @@ func (c *controller) updateTagsFromFile(ctx context.Context,
 	}
 	return nil
 }
+
+// Deprecated: for internal usage
+// manifestVersionChanged determines whether manifest version is changed
+func (c *controller) manifestVersionChanged(ctx context.Context, application,
+	cluster string, commit string) (bool, error) {
+	currentManifest, err1 := c.clusterGitRepo.GetManifest(ctx, application, cluster, nil)
+	targetManifest, err2 := c.clusterGitRepo.GetManifest(ctx, application, cluster, &commit)
+	for _, err := range []error{err1, err2} {
+		if err != nil {
+			if _, ok := perror.Cause(err).(*herrors.HorizonErrNotFound); !ok {
+				log.Errorf(ctx, "get cluster manifest error, err = %s", err.Error())
+				return false, err
+			}
+		}
+	}
+	if err1 != nil && err2 != nil {
+		// both v1
+		return false, nil
+	}
+	if err1 != nil || err2 != nil {
+		// v1, v2
+		return true, nil
```

Comment:
why it returns true when there's an err found?

Reply:
here, the err is caused by herrors.HorizonErrNotFound, which shows the manifest does not exist (v1)